ddtrace/contrib/aiobotocore/patch.py

```diff
@@ -0,0 +1,158 @@
+            result = yield from self.__wrapped__.read(*args, **kwargs)  # noqa: E999
+            span.set_tag('Length', len(result))
+
+        return result
+
+    if PY_VER >= (3, 5, 0):
+        @asyncio.coroutine
+        def __aenter__(self):
+            result = yield from self.__wrapped__.__aenter__()  # noqa: E999
+            assert result == self.__wrapped__
+            return self
+
+        @asyncio.coroutine
+        def __aexit__(self, *args, **kwargs):
+            result = yield from self.__wrapped__.__aexit__(*args, **kwargs)  # noqa: E999
+            return result
+
+
+def truncate_arg_value(value, max_len=1024):
+    """ Method will truncate values which are bytes and greater than `max_len`.
+    Useful for parameters like 'Body' in `put_object` operations. """
+    if isinstance(value, bytes) and len(value) > max_len:
+        return b'...'
+
+    return value
+
+
+@asyncio.coroutine
+def patched_api_call(original_func, instance, args, kwargs):
+    pin = Pin.get_from(instance)
+    if not pin or not pin.enabled():
+        result = yield from original_func(*args, **kwargs)  # noqa: E999
+        return result
+
+    endpoint_name = deep_getattr(instance, "_endpoint._endpoint_prefix")
+
+    with pin.tracer.trace('{}.command'.format(endpoint_name),
+                          service="{}.{}".format(pin.service, endpoint_name),
+                          span_type=SPAN_TYPE) as span:
+
+        # set parent trace/span IDs if present:
+        #    http://pypi.datadoghq.com/trace/docs/#distributed-tracing
+        parent_trace_id = args[1].pop(PARENT_TRACE_KWARG_ID, None)
+        if parent_trace_id is not None:
+            span.trace_id = int(parent_trace_id)
+
+        parent_span_id = args[1].pop(PARENT_SPAN_KWARG_ID, None)
+        if parent_span_id is not None:
+            span.parent_id = int(parent_span_id)
+
+        operation = None
+        if args:
+            operation = args[0]
+            span.resource = '%s.%s' % (endpoint_name, operation.lower())
+
+        else:
+            span.resource = endpoint_name
+
+        # Adding the args in TRACED_ARGS if exist to the span
+        if not aws.is_blacklist(endpoint_name):
+            for name, value in aws.unpacking_args(args, ARGS_NAME, TRACED_ARGS):
+                if name == 'params':
+                    value = {k: truncate_arg_value(v) for k, v in value.items()}
+                span.set_tag(name, (value))
+
+        region_name = deep_getattr(instance, "meta.region_name")
+
+        meta = {
+            'aws.agent': 'aiobotocore',
+            'aws.operation': operation,
+            'aws.region': region_name,
+        }
+        span.set_tags(meta)
+
+        result = yield from original_func(*args, **kwargs)  # noqa: E999
+
+        body = result.get('Body')
+        if isinstance(body, ClientResponseContentProxy):
+            result['Body'] = WrappedClientResponseContentProxy(body, pin, span)
+
```

Comment:
(2) We may also do without the WrappedClient extra span if making a test is complex